src/onegov/form/parser/core.py

```diff
@@ -1188,6 +1188,8 @@ def parse_formcode(
             parse_field_block(block, field_classes, used_ids, fs)
             for block in (fieldset[label] or ())
         ]
+        if not fs.fields:
+            raise errors.EmptyFieldsetError(label)
```

Comment:
I'm a little bit concerned about putting this here. Since there almost certainly will be old forms that have this issue and rather than continue to work, they will now stop working. If you put this here you also need to write a migration to clean out all the empty field sets from existing form definitions.

For now it might be easier to make this check optional, just like the indent check. Then you can only enable it within the validator. Alternatively you can do the check entirely in the validator by iterating over the parsed form's fieldsets.

Reply:
Valid point!

Reply:
done